src/routes/news/components/NewsList.js

```diff
@@ -0,0 +1,264 @@
+  font-family: ${props => props.theme.fonts.contentFontRegular};
+  align-self: center;
+  margin-top: 20px;
+`
+
+const NewsHeadLine: StyledComponent<{}, ThemeType, *> = styled.Text`
+  font-weight: 700;
+  font-family: ${props => props.theme.fonts.decorativeFontBold};
+  color: ${props => props.theme.colors.textColor};
+  font-size: 18px;
+  margin-top: 18px;
+  margin-bottom: 15px;
+`
+
+const NewsContent: StyledComponent<{ language: string }, ThemeType, *> = styled.Text`
+  font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  font-size: 16px;
+  letter-spacing: 0.5px;
+  line-height: 24px;
+  text-align: ${props => contentAlignment(props.language)};
+  color: ${props => props.theme.colors.textColor};
+`
+export type PropsType = {|
+  newsId: ?string,
+  news: NewsModelsType,
+  cityCode: string,
+  language: string,
+  resourceCache: LanguageResourceCacheStateType,
+  theme: ThemeType,
+  t: TFunction,
+  selectedNewsType: NewsType,
+  status: "ready" | "loadingMore",
+  isFetchingMore: boolean,
+  fetchMoreNews: () => void,
+  navigateToNews: (navigationOptions: NavigateToNewsParamsType) => void,
+  createNavigateToNews: (NavigateToNewsParamsType) => void
+|}
+
+/**
+ * Displays a list of news or a single news item, matching the route <id>)
+ */
+class NewsList extends React.PureComponent<PropsType> {
+  navigateToNews = (cityCode: string, language: string, newsId: string) => () => {
+    const { selectedNewsType } = this.props
+    this.props.navigateToNews({
+      cityCode,
+      language,
+      newsId,
+      type: selectedNewsType
+    })
+  };
+
+  openTunewsLink = async () => {
+    const supported = await Linking.canOpenURL(tunewsWebsiteUrl)
+    if (supported) {
+      await Linking.openURL(tunewsWebsiteUrl)
+    }
+  };
+
+  renderNoItemsComponent = () => {
+    const { t, theme } = this.props
+    return <NoNews theme={theme}>{t('currentlyNoNews')}</NoNews>
+  };
+
+  rendersNewsListItem = (cityCode: string, language: string) => ({
+    item: newsItem
+  }) => {
+    const { theme, selectedNewsType } = this.props
+    const isTunews = selectedNewsType === TUNEWS
+    return (
+      <NewsListItem
+        key={newsItem.id}
+        newsItem={newsItem}
+        language={language}
+        theme={theme}
+        isTunews={isTunews}
+        navigateToNews={this.navigateToNews(cityCode, language, newsItem.id.toString())}
+      />
+    )
+  };
+
+  render () {
+    const {
+      news,
+      newsId,
+      cityCode,
+      language,
+      theme,
+      t,
+      fetchMoreNews,
+      isFetchingMore,
+      selectedNewsType
+    } = this.props
+
+    const isTunews = selectedNewsType === TUNEWS
+
+    if (newsId) {
+      const selectedNewsItem: LocalNewsModel | TunewsModel | typeof undefined = news.find(
+        _newsItem => _newsItem.id.toString() === newsId
+      )
+
+      if (selectedNewsItem) {
+        const isInstanceOfTunews = selectedNewsItem instanceof TunewsModel
```

Comment:
It would be better to have two separate NewsLists. Eg. LocalNewsList or TüNewsList.
In order to avoid duplicating code renderProps or passing children could be used: https://reactjs.org/docs/render-props.html

Using `instanceof` smells like polymorphims was misused.

Reply:
I deleted the `instanceof`, made some refactor. IMO its a bit complicated to use renderProps here. See my fix and let me know if it needs change